### PR TITLE
Allow Cassandra passthrough to be configured in docker-compose.yml 

### PR DESCRIPTION
**Pull request by conorbros**
This change allows us to use docker compose volume mounting for files the example directory. 

E.g. We can just mount the PWD in `docker-compose.yml`

```YAML
volumes:
    - ./docker-entrypoint.sh:/usr/local/bin/docker-entrypoint.sh
```

I've also included the change to the `cassandra-passthrough` example that I needed this change for. User defined functions need to be enabled in Cassandra to test them but they can't be set from the docker-compose so we need to modify the `docker-entrypoint.sh` to do this.

The `docker-entrypoint.sh` is mostly the same as the [default](https://github.com/docker-library/cassandra/blob/master/3.11/docker-entrypoint.sh) but I added `enable_user_defined_functions` and `enable_scripted_user_defined_functions` to the `for yaml in \` loop.

I considered opening a PR to enable this on the official docker image but I found this [comment](https://github.com/docker-library/cassandra/pull/122#issuecomment-426774071) explaining why they don't want to enable this.

### CODE REVIEW

File: shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh
Comment (benbromhead) [inline shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh:1]: this file looks like we've taken it from the docker image we pull in. Do you want to check whether we need to add anything to our NOTICES file from a license perspective?

**Review by benbromhead — APPROVED** [review]

File: shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh
Comment (rukai) [inline shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh:2]: Can you also add a comment at the top describing how this differs from the default file?

File: shotover-proxy/examples/cassandra-passthrough/docker-compose.yml
Comment (rukai) [inline shotover-proxy/examples/cassandra-passthrough/docker-compose.yml:14]: Cant we just specify the path here relative to the default CWD instead of changing the CWD?

**Review by rukai — CHANGES_REQUESTED** [review]

File: shotover-proxy/examples/cassandra-passthrough/docker-compose.yml
Comment (conorbros) [reply shotover-proxy/examples/cassandra-passthrough/docker-compose.yml:14]: I frequently run the examples from within the directory and I would assume users will attempt the same. Also if you just download the example files to run them the paths won't be right.

Comment (rukai) [reply shotover-proxy/examples/cassandra-passthrough/docker-compose.yml:14]: ah! good point

Comment (rukai) [reply shotover-proxy/examples/cassandra-passthrough/docker-compose.yml:14]: Wait, uh... how does the redis-passthrough work?
It is already doing the same volume thing.

File: shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh
Comment (conorbros) [reply shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh:1]: @justinmclean

File: shotover-proxy/examples/cassandra-passthrough/docker-compose.yml
Comment (conorbros) [reply shotover-proxy/examples/cassandra-passthrough/docker-compose.yml:14]: ...interesting. I thought the volume mounting worked relative to the processes CWD not the directory of the `docker-compose.yml`.

Comment (conorbros) [reply shotover-proxy/examples/cassandra-passthrough/docker-compose.yml:14]: Reverted the changes.

File: shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh
Comment (conorbros) [reply shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh:2]: Addressed this btw

**Review by rukai — APPROVED** [review]
This can be merged once you've resolved the licensing issues.
One day we might want to change this to directly provide the yaml file instead of generating it in bash.
But for now this is fine.

**Comment by conorbros** [comment]
Sounds good. BTW it won't merge until that  conversation is resolved so it's safe to enable auto-merge.

File: shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh
Comment (conorbros) [reply shotover-proxy/examples/cassandra-passthrough/docker-entrypoint.sh:1]: Added the changes to LICENSE

**Pull request merged**
